lambdas/format_input.py

```diff
@@ -66,8 +68,10 @@ def lambda_handler(event: dict, _context: dict) -> dict:
             source,
         )
         result["next-step"] = "load"
+
+        # NOTE: FEATURE FLAG: branching logic will be removed after v2 work is complete
         result["transform"] = commands.generate_transform_commands(
-            extract_output_files, event, run_date, timdex_bucket
+            extract_output_files, event, run_date, timdex_bucket, etl_version
```

Comment:
What's the thinking behind retrieving the `ETL_VERSION` here in the main handler, vs retrieving that value within `generate_transform_commands()`?  

If we pulled this value only there, it seems like we could avoid:
1. changing the overall signature of `generate_transform_commands()`
2. basically leave this `lambda_handler()` function untouched

Curious your thoughts!

Reply:
Ah! Good question. I noticed in Transmogrifier, the variable was retrieved at the beginning of `transmogrifier.cli.main`, and it felt like retrieving it at the start of the `lambdas.format_input.lambda_handler` was the equivalent! 🤔 I also figured if, for whatever reason, the provided `ETL_VERSION` was not valid, it would stop the function early.

Given that the branching logic affects only the Transform and Load command and has no impact on the Extract command, I do think it would make sense to call `get_etl_version` inside these functions instead and avoiding changing the signature of the 'generate' commands. 

Let me know what you think given this additional info and curious what @ehanson8 thinks as well on how to do this as smoothly as possible!

Reply:
The choice to do [that here in Transmogrifier](https://github.com/MITLibraries/transmogrifier/blob/main/transmogrifier/cli.py#L64-L70), was to allow jumping into a completely different "entrypoint" method for the Transformer class.  

As I type this, I can see how that pattern could also be applied here?  For example, instead of still calling `generate_transform_command()`, we could have two helper functions defined:

1. `generate_transform_command_v1()`
2. `generate_transform_command_v2()`

In this scenario, no function signatures are really changed, you just branch from the main `lambda_handler()` function.

That might be a cleaner approach than what I've kind of proposed in the comments:

1. keep `lambda_handler()` function untouched
2. enter `generate_transform_command()` and get `ETL_VERSION`
3. call either `_etl_v1_generate_transform_commands_method` or `_etl_v2_generate_transform_commands_method`

Ultimately, I think any of them work.  I'd approach the decision from the POV of what is easiest to remove later, and what allows functions and signatures to remain largely the same where possible.  The second item there -- function signatures -- is where I think passing the `etl_version` might be a bit more disruptive than it needs to be.

Reply:
Thank you for this additional information and providing a very helpful perspective!

I went with the original suggestion of moving the `get_etl_version()` calls to the 'generate' commands for the Transform and Load step. See the last two commits!